### PR TITLE
project config and mux interfaces

### PR DESCRIPTION
**Pull request by jrhea**
adding base mux interfaces and gradle files

### CODE REVIEW

**Comment by raulk** [comment]
BTW – gonna try to adopt the Kotlin DSL for Gradle, if you guys agree. I personally have a pet peeve with Groovy 😛

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,233 @@
+
+import net.ltgt.gradle.errorprone.CheckSeverity
+
+plugins {
```
Comment (raulk) [inline build.gradle]: Gonna drop most of these plugins until we actually need them – really want to focus on not bloating the build (I've probably spent too much time on Go now :)).

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  id 'com.diffplug.gradle.spotless' version '3.16.0'
+  id 'net.ltgt.errorprone' version '0.6'
```
Comment (raulk) [inline build.gradle]: I don't think errorprone works with kotlin.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,233 @@
+
+import net.ltgt.gradle.errorprone.CheckSeverity
+
+plugins {
+  id 'com.diffplug.gradle.spotless' version '3.16.0'
```
Comment (raulk) [inline build.gradle]: We probably want to adopt https://github.com/pinterest/ktlint instead.

File: build.gradle (hunk 2 above)
Comment (jrhea) [reply build.gradle]: removed

File: build.gradle (hunk 3 above)
Comment (jrhea) [reply build.gradle]: switched to ktlint.  still needs to be configured

File: build.gradle (hunk 1 above)
Comment (jrhea) [reply build.gradle]: this sounds personal.  feel free to satisfy your OCD by removing whatever you feel is necessary.  i will usually just go with the flow

File: mux/src/main/kotlin/io/libp2p/mux/MuxedStream.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    fun close(): Boolean
+
+    fun reset()
+
+    fun setDeadline(date: Date)
+
+    fun setReadDeadline(date: Date)
+
+    fun setWriteDeadcline(date: Date)
```
[the remark this answers is not in the document]
Comment (jrhea) [reply mux/src/main/kotlin/io/libp2p/mux/MuxedStream.kt]: resolved

File: mux/src/main/kotlin/io/libp2p/mux/MuxedStream.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.Date
+import org.apache.tuweni.bytes.Bytes
+/**
+ * MuxedStream is a bidirectional io pipe within a connection.
+ * Adapted from Go reference implementation:
+ * https://github.com/libp2p/go-libp2p-core/blob/master/mux/mux.go
+ */
+interface MuxedStream {
+
+    fun read(): Bytes
```
[the remark this answers is not in the document]
Comment (jrhea) [reply mux/src/main/kotlin/io/libp2p/mux/MuxedStream.kt]: It is essentially a placeholder.  just trying to jumpstart this effort.

**Pull request closed without merging**